lib/utils/log/formatter_test.go

```diff
@@ -0,0 +1,381 @@
+	"github.com/stretchr/testify/require"
+)
+
+const message = "Adding diagnostic debugging handlers.\t To connect with profiler, use `go tool pprof diag_addr`."
+
+var (
+	logErr = errors.New("the quick brown fox jumped really high")
+	addr   = fakeAddr{addr: "127.0.0.1:1234"}
+
+	fields = logrus.Fields{
+		"local":        &addr,
+		"remote":       &addr,
+		"login":        "llama",
+		"teleportUser": "user",
+		"id":           1234,
+	}
+)
+
+type fakeAddr struct {
+	addr string
+}
+
+func (a fakeAddr) Network() string {
+	return "tcp"
+}
+
+func (a fakeAddr) String() string {
+	return a.addr
+}
+
+func TestOutput(t *testing.T) {
+	// Set a specific time zone so that the regex doesn't have to match
+	// against any possible timezone. Note this mucks around with the global
+	// time zone which prevents running this test in parallel.
+	loc, err := time.LoadLocation("Africa/Cairo")
+	require.NoError(t, err, "failed getting timezone")
+	oldLoc := time.Local
+	time.Local = loc
+	t.Cleanup(func() {
+		time.Local = oldLoc
+	})
+
+	t.Run("text", func(t *testing.T) {
+		// fieldsRegex matches all the key value pairs emitted after the message and before the caller. All fields are
+		// in the following format key:value key2:value2 key3:value3.
+		fieldsRegex := regexp.MustCompile(`(\w+):((?:"[^"]*"|\[[^]]*]|\S+))\s*`)
+		// outputRegex groups the entire log output into 4 distinct matches. The regular expression is tailored toward
+		// the output emitted by the test and is not a general purpose match for all output emitted by the loggers.
+		// 1) the timestamp, component and level
+		// 2) the message
+		// 3) the fields
+		// 4) the caller
+		outputRegex := regexp.MustCompile("(\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\+\\d{2}:\\d{2})(\\s+.*)(\".*diag_addr`\\.\")(.*)(\\slog/formatter_test.go:\\d{3})")
+
+		tests := []struct {
+			name        string
+			logrusLevel logrus.Level
+			slogLevel   slog.Level
+		}{
+			{
+				name:        "trace",
+				logrusLevel: logrus.TraceLevel,
+				slogLevel:   TraceLevel,
+			},
+			{
+				name:        "debug",
+				logrusLevel: logrus.DebugLevel,
+				slogLevel:   slog.LevelDebug,
+			},
+			{
+				name:        "info",
+				logrusLevel: logrus.InfoLevel,
+				slogLevel:   slog.LevelInfo,
+			},
+			{
+				name:        "warn",
+				logrusLevel: logrus.WarnLevel,
+				slogLevel:   slog.LevelWarn,
+			},
+			{
+				name:        "error",
+				logrusLevel: logrus.ErrorLevel,
+				slogLevel:   slog.LevelError,
+			},
+			{
+				name:        "fatal",
+				logrusLevel: logrus.FatalLevel,
+				slogLevel:   slog.LevelError + 1,
+			},
+		}
+
+		for _, test := range tests {
+			t.Run(test.name, func(t *testing.T) {
+				// Create a logrus logger using the custom formatter which outputs to a local buffer.
+				var logrusOutput bytes.Buffer
+				formatter := NewDefaultTextFormatter(true)
+				formatter.timestampEnabled = true
+				require.NoError(t, formatter.CheckAndSetDefaults())
+
+				logger := logrus.New()
```

Comment:
Nit: I'd probably call this `logrusLogger` or similar - just to distinguish it clearly from `slogLogger`.